### PR TITLE
fix: #1481 Allow frontend verifying test user accounts in FAM PROD for TEST apps

### PR DESCRIPTION
**Pull request by MCatherine1994**
refs: #1481

- Backend IDIM web proxy router will use the IDIM web url based on the given app environment
- Router guard for verifying target user need to use the IDIM web url based on app environment

### CODE REVIEW

File: terraform/prod/terragrunt.hcl [hunk 1]
```diff
@@ -40,5 +40,7 @@ generate "prod_tfvars" {
   forest_client_api_base_url = "https://nr-forest-client-api-prod.api.gov.bc.ca/"
   use_override_proxy_endpoints = false
   idim_proxy_api_base_url = "https://nr-fam-idim-lookup-proxy-prod-backend.apps.silver.devops.gov.bc.ca"
+  idim_proxy_api_base_url_prod = "https://nr-fam-idim-lookup-proxy-prod-backend.apps.silver.devops.gov.bc.ca"
```
Comment (MCatherine1994) [inline terraform/prod/terragrunt.hcl:43]: only FAM PROD will need production url, only PROD APP in FAM PROD can verify production users

File: server/backend/api/app/integration/idim_proxy.py [hunk 2]
```diff
@@ -20,10 +19,11 @@ class IdimProxyService:
 
     TIMEOUT = (5, 10)  # Timeout (connect, read) in seconds.
 
-    def __init__(self, requester: Requester):
+    def __init__(self, requester: Requester, app_env: AppEnv = AppEnv.APP_ENV_TYPE_TEST):
```
Comment (MCatherine1994) [inline server/backend/api/app/integration/idim_proxy.py:22]: set default app_env to be TEST, so it won't impact our tests

File: terraform/prod/terragrunt.hcl [hunk 3]
```diff
@@ -40,5 +40,7 @@ generate "prod_tfvars" {
   forest_client_api_base_url = "https://nr-forest-client-api-prod.api.gov.bc.ca/"
   use_override_proxy_endpoints = false
   idim_proxy_api_base_url = "https://nr-fam-idim-lookup-proxy-prod-backend.apps.silver.devops.gov.bc.ca"
```
Comment (MCatherine1994) [inline terraform/prod/terragrunt.hcl:42]: Admin management apis are still using this variable, we could remove it once we figure out how to update the router guard to verify users with IDIM url based on app env

File: server/backend/api/app/integration/idim_proxy.py [hunk 4]
```diff
@@ -20,10 +20,19 @@ class IdimProxyService:
...
+        # TODO: get_idim_proxy_api_baseurl() already has swithcing logic based on app_env
+        # comment below by passing app_env to config; review again later.
+        # if app_env is None or app_env == AppEnv.APP_ENV_TYPE_DEV:
+        #     self.api_instance_env = AppEnv.APP_ENV_TYPE_TEST
```
Comment (MCatherine1994) [inline server/backend/api/app/integration/idim_proxy.py:27]: Hi Ian, just wondering do we still need this api_instance_env comment?

**Comment by MCatherine1994** [comment]
thanks so much Ian!! Really good fix, just a small comment, but looks good to me.

**Review by MCatherine1994 — COMMENTED** [review]
looks good

**Review by ianliuwk1019 — APPROVED** [review]

**Pull request merged**